README.md

```diff
@@ -1,17 +1,269 @@
+  display_server: {:port, display}
+])
+```
+
+## Display Lists
+
+The display list is a declarative representation of what should be shown on screen. It consists of
+primitive drawing elements that are rendered in order (last item is drawn first, first item appears
+on top).
+
+### Supported Primitives
+
+- `{rect, X, Y, Width, Height, Color}` - Filled rectangle
+- `{text, X, Y, Font, TextColor, BackgroundColor, Text}` - Text rendering
+- `{image, X, Y, BackgroundColor, ImageTuple}` - Image display
+- `{scaled_cropped_image, ...}` - Scaled and cropped image
+
+For complete documentation on display lists and primitives, see the
+[AtomGL documentation](https://github.com/atomvm/atomgl).
+
+## Input Handling
+
+To handle input events (touch, buttons, etc.), implement the optional `handle_input/4` callback:
+
+### Erlang
+
+```erlang
+handle_input(EventData, Timestamp, Pid, State) ->
+    io:format("Input event: ~p at ~p~n", [EventData, Timestamp]),
+    {noreply, State}.
 ```
+
+### Elixir
+
+```elixir
+def handle_input(event_data, timestamp, pid, state) do
+  IO.puts("Input event: #{inspect(event_data)} at #{timestamp}")
+  {:noreply, state}
+end
+```
+
+Then provide an `input_server` when starting the scene:
+
+```erlang
+{ok, Pid} = hello_scene:start_link([], [
+    {display_server, {port, Display}},
+    {input_server, InputServerPid}
+]).
+```
+
+## Advanced Example: Interactive Counter
+
+Here's a more complete example showing state management and display updates:
+
+### Elixir
+
+```elixir
+defmodule CounterScene do
+  def start_link(args, opts) do
+    :avm_scene.start_link(__MODULE__, args, opts)
+  end
+
+  def init(_args) do
+    # Initial render
+    self() |> send(:render)
+
+    {:ok, %{
+      counter: 0,
+      width: 320,
+      height: 240
+    }}
+  end
+
+  def handle_info(:increment, state) do
+    new_state = %{state | counter: state.counter + 1}
+    self() |> send(:render)
+    {:noreply, new_state}
+  end
+
+  def handle_info(:render, state) do
+    items = [
+      # Counter display
+      {:text, 100, 100, :default16px, 0x000000, :transparent,
+       "Count: #{state.counter}"},
+
+      # Increment button
+      {:rect, 50, 150, 100, 40, 0x4444FF},
+      {:text, 70, 165, :default16px, 0xFFFFFF, :transparent, "Increment"},
+
+      # Background
+      {:rect, 0, 0, state.width, state.height, 0xFFFFFF}
+    ]
+
+    {:noreply, state, [{:push, items}]}
+  end
+
+  def handle_input({:touch, x, y}, _timestamp, _pid, state) do
+    # Check if increment button was pressed
+    if x >= 50 and x <= 150 and y >= 150 and y <= 190 do
+      self() |> send(:increment)
+    end
+
+    {:noreply, state}
+  end
+end
+```
+
+## Configuration Options
+
+When starting an `avm_scene`, you can provide the following options:
+
+| Option | Type | Required | Description |
+|--------|------|----------|-------------|
+| `display_server` | `{Module, Display}` | Yes | Display driver reference, typically `{:port, DisplayPort}` |
```

Comment:
The description clarifies this, but the use of `Module` seems misleading here, shouldn't the first element always be a `port` atom?